### DISCUSSION TITLE
How to ignore files by pattern?

### DISCUSSION THREAD

**Discussion by ghost** [nvim-tree/nvim-tree.lua#2091] [Q&A]
Null-ls constantly creates its temporary files in the buffer directory, and it looks a bit ugly and distracting.

https://user-images.githubusercontent.com/107887698/227794379-f6cf670a-7106-4535-afac-7676199a3dac.mp4

(The video is a bit old and not mine)

How can I tell nvim-tree to ignore these files?

**Comment by alex-courtis** [comment]
I occasionally see that with some clang operations.

See `:help nvim-tree.filters.custom`

**Reply by ghost** [reply]
Yes, I read that and tried to do something, but it didn't work. Can you show me an example of what to put in the `custom`?

**Reply by alex-courtis** [reply]
It's a vim regex: `:help string-match`

I use
```lua
  filters = {
    custom = {
      "^.git$",
    },
  },
```

**Reply by ghost** [reply]
Am I doing the right thing?

```
  filters = {
    custom = {
      "^.null-ls_*$",
    },
  },
```

**Reply by alex-courtis** [reply]
I don't know what your file name is, however I think you might want `"^.*null-ls_.*$"`

Experiment:
```
:lua print(vim.fn.match("foo-null-ls_bar", "^.*null-ls_.*$"))
```

**Reply by ghost** [reply]
Only the command works, i.e. when I enter this template into "custom", Nvim-tree does nothing (or I'm doing something wrong). When I enter this command, and there is this file in the directory, it outputs "-1".
   Null-ls creates files of the type ".null-ls_00000_test.py".

**Reply by alex-courtis** [reply]
`-1` indicates no match. You'll need to tweak the regex; I just took a guess. `:help match()`

Please also note that you need to be mindful of any characters that require escaping in a lua string.

Suggestion: start small: a very basic regex like `".*null"` and build it up from there.
